lib/ipfs/RequestHandler.ts

```diff
@@ -18,13 +18,28 @@ export default class RequestHandler {
 
   private serviceInfo: ServiceInfo;
 
+  /**
+   * create an instance of request handler
+   * @param fetchTimeoutInSeconds Timeout for fetch request. Fetch request will return `not-found` when timed-out.
+   * @param repo Optional IPFS datastore implementation.
+   */
+  public static async create (fetchTimeoutInSeconds: number, repo?: any) {
+    let ipfsStorage: IpfsStorage;
+    try {
+      ipfsStorage = await IpfsStorage.createSingleton(repo);
+    } catch {
+      ipfsStorage = IpfsStorage.getSingleton();
```

Comment:
We should throw a specific exception-error-code in createSingleton and only call getSingleton() if that exception is thrown. Other exceptions should be just bubbled up.